codalab/worker/bundle_state.py

```diff
@@ -31,3 +31,52 @@ class State(object):
     OPTIONS = {CREATED, STAGED, MAKING, STARTING, RUNNING, READY, FAILED, PREPARING, FINALIZING}
     ACTIVE_STATES = {MAKING, STARTING, RUNNING, FINALIZING, PREPARING}
     FINAL_STATES = {READY, FAILED, KILLED}
+
+
+class RunResources(object):
+    """
+    Defines all the resource fields the server propagates to the worker for its runs
+    """
+
+    def __init__(self, cpus, gpus, docker_image, time, memory, disk, network):
```

Comment:
might be a good idea to start inserting type annotations in new code

Reply:
Hmm, I've been slowly adding comment-annotations to keep py2 compatibility. I guess won't hurt to keep doing that.

Reply:
Doing this in another PR that replaces a few more of these unstructured dicts with objects